### PR TITLE
gettext, texinfo: Remove spurious xz 

### PR DESCRIPTION
**Pull request by rhelmot**
## Description of changes

part of https://github.com/NixOS/nixpkgs/pull/296581

If xz is overridden with a derivation that does not have an explicit bin output, these will fail. These changed instances are the ones impacted by the FreeBSD stdenv providing a bootstrap-files xz for early boot.

diffoscope indicates that the builds with and without xz are identical, so just remove.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
  - [x] x86_64-freebsd
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by wolfgangwalther — APPROVED** [review]

File: pkgs/development/libraries/gettext/default.nix [hunk 1]
```diff
@@ -52,7 +52,7 @@ stdenv.mkDerivation rec {
   strictDeps = true;
   nativeBuildInputs = [
     xz
-    xz.bin
+    (lib.getBin xz)
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gettext/default.nix:55]: Isn't this automatically chosen here?

File: pkgs/development/tools/misc/texinfo/common.nix [hunk 2]
```diff
@@ -49,7 +49,7 @@ stdenv.mkDerivation {
   # A native compiler is needed to build tools needed at build time
   depsBuildBuild = [ buildPackages.stdenv.cc perl ];
 
-  buildInputs = [ xz.bin bash libintl ]
+  buildInputs = [ (lib.getBin xz) bash libintl ]
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/texinfo/common.nix:52]: This seems wrong to me especially for cross compilation

Comment (Ericson2314) [reply pkgs/development/tools/misc/texinfo/common.nix:52]: I think @SuperSandro2000 means this should be a `nativeBuildInput`

**Review by Ericson2314 — CHANGES_REQUESTED** [review]
Please put the description in the commit messages (goes for many of the other PRs too).

I think it's OK to just leave the `texinfo` one as-is with the suspicious use of `buildInputs` as that's an orthogonal problem, but if you are feeling motivated you could add a `strictDeps = true` and see if `nativeBuildInputs` is better (simulating cross).

File: pkgs/development/tools/misc/texinfo/common.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/misc/texinfo/common.nix:52]: > Is your comment about `xz.bin` in the first place

yes

Comment (SuperSandro2000) [reply pkgs/development/tools/misc/texinfo/common.nix:52]: We don't need that here anymore. At the time the tarball was in the xz format but now we have gz.

File: pkgs/development/libraries/gettext/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/gettext/default.nix:55]: We don't need that here anymore. At the time the tarball was in the xz format but now we have gz and strictDeps is already set.

```suggestion
  buildInputs = [ bash libintl ]
```

The --with-xz seems to be not properly checked and for that to work we would need to move xz to buildInputs

File: pkgs/development/tools/misc/texinfo/common.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/misc/texinfo/common.nix:52]: We don't need that here anymore. At the time the tarball was in the xz format but now we have gz and strictDeps is already set.

File: pkgs/development/libraries/gettext/default.nix (hunk 1 above)
Comment (SuperSandro2000) [inline pkgs/development/libraries/gettext/default.nix:55]:
```suggestion
```
I've run diffoscope on a gettext build with and without xz and booth are equal. I think we can just drop xz entirely from here.

and we can also drop the ``"--with-xz"`` arg in line 31.

**Comment by rhelmot** [comment]
I have changed the commits as per feedback, removing both uses!

**Review by Ericson2314 — APPROVED** [review]

**Pull request merged**